### PR TITLE
Add new method to detect code (resolves #2)

### LINKED ISSUE #2: Add new method to detect code

### PR DESCRIPTION
**Pull request by vidhanio**
Uses context menus in messages now.
image

### CODE REVIEW

File: bot.go [hunk 1]
```diff
@@ -29,289 +27,337 @@ var (
...
-	cropped := len(toCrop) - end
-	end -= int(math.Log10(float64(cropped))) + 1 // remove estimate of number of characters that the number takes
-	if toCrop[end-2] == '/' && toCrop[end-3] != '/' {
-		end -= 2 // remove whatever is the last
-	} else if toCrop[end-1] == '/' {
-		end--
-	}
-	return toCrop[:end], len(toCrop) - end // recalculated cropped
-}
+	// Remove the 6 backticks and 2 newlines from the limit.
```
Comment (Nathan13888) [inline bot.go:623]: are there tabs/spaces in front of these?

**Review by Nathan13888 — APPROVED** [review]
Looks good. Haven't tested it but prob works.

**Comment by Nathan13888** [comment]
## Changes
* improved logging
* better processing on environment variables
* improved comments
* implementation of SLASH commands
* new method for running code
* 

## Things I would change
* [x] add debug logging for settings in `init()`
* [x] add back bot status
```
dg.AddHandler(func(s *discordgo.Session, event *discordgo.Ready) {
	s.UpdateGameStatus(0, "!run")
})
```
* [x] reimplement `build` (version)
* [x] reimplement `help` command

**Review by Nathan13888 — APPROVED** [review]
Looks good to me. Can merge :)

**Comment by vidhanio** [comment]
🚀🚀🚀🚀🚀

**Pull request merged**